general/db/guide-gigaspaces.asciidoc

```diff
@@ -0,0 +1,47 @@
+:toc: macro
+toc::[]
+
+= GigaSpaces XAP (Smart Cache)
+
+This section is the place to share experience for those who use https://www.gigaspaces.com/products/xap/[GigaSpaces XAP] as link:guide-database.asciidoc#nosql[NoSQL database].
+
+== Attention
+NOTE: devonfw did not focus on the integration of this database so far. No reports have been given from our users about successfully integrating this database with any devonfw tech stack. If you want to share your knowledge or report usage, please contribute by clicking on the pen next to the section headline.If you need help on devonfw tech stack knowledge to get the integration working for you, stay in contact at https://github.com/devonfw/devonfw-guide/issues[GitHub.]
+
+== Possible Approach
+GigaSpaces is currently not in the central maven repository, therefore an additional repository needs to be added along with the dependency:
+```
+<repositories>
+    <repository>
+	<id>org.openspaces</id>
+	<url>http://maven-repository.openspaces.org</url>
+    </repository>
+</repositories>
+
+<dependency>
+    <groupId>org.gigaspaces</groupId>
+    <artifactId>xap-openspaces</artifactId>
+    <version>${gsVersion}</version>
+</dependency>
+```
+Of course the version (`${gsVersion}`) needs to be adopted to your needs.
+
+[source, java]
+----
+@Configuration
+public class ContextConfiguration {
+  @Bean
+  public GigaSpace space() {
+    UrlSpaceConfigurer urlSpaceConfigurer = new UrlSpaceConfigurer("jini://*/*/my-space");
+    return new GigaSpaceConfigurer(urlSpaceConfigurer).gigaSpace();
+  }
+}
+----
+
+To establish a connection with a running istance of GigaSpaces, a Configuration Class is required. Here a Bean will be created that retrieves via URL the name of a Space e.g. my-space (a Space is equivalent to a Database Schema). Of course a Space needs to be firstly created in order to use it (see also the Example). This bean can be used for all database typical operations e.g. create, read, update and delete data (a complete list can be found https://resources.gigaspaces.com/api/15.8.0/java/org/openspaces/core/GigaSpace.html[here]). Another possibility to execute those operations is via spring-data (see section below). The spring-data-gigaspaces automatically detects if a GigaSpaces Bean exists.

```

Comment:
see also the "Example" should be a xref to the example section. Otherwise, someone might easily ask and start searching the example, which is not discussed at all in this section.